draft-dt-tls-external-psk-guidance.md

```diff
@@ -274,6 +256,22 @@ pair-wise shared keys to achieve this. As another example, some systems require
 application-specific information in either PSKs or their identities. Identities may sometimes need to be routable,
 as is currently under discussion for EAP-TLS-PSK.
 
+
+# Recommendations for External PSK Usage
+
+
+Applications MUST use external PSKs that adhere to the following requirements:
+
+1. Each PSK MUST be derived from at least 128 of entropy and MUST be at least 128-bits long unless the TLS handshake is being used with a separate key establishment mechanism such as a Diffie-Hellman exchange. This recommendation protects against passive attacks using exhaustive search of the PSK.
+2. Each PSK MUST NOT be shared between with more than two logical nodes. As a result, an agent that acts as both a client and a server MUST use distinct PSKs when acting as the client from when it is acting as the server. This prevents redirection attacks.
+3. Nodes SHOULD use external PSK importers {{!I-D.ietf-tls-external-psk-importer}} when configuring PSKs for a pair of TLS client and server. {{!RFC8446}} mandates that each external PSK can only be used with a single hash function. {{!I-D.ietf-tls-external-psk-importer}} overcomes this limitation by producing a set of candidate PSKs, each of which are 
+bound to a specific KDF. 
+4. {{!I-D.ietf-tls-external-psk-importer}} also helps in mitigating selfie style reflection attacks if a pair of client and server can switch roles. This is achieved by correctly using the node identifiers in the ImportedIdentity.context construct as explained in Appendix B of {{!I-D.ietf-tls-external-psk-importer}}. When a PSK is shared between a client and server that can switch roles, it is not necessary to know the identifier(s) of the other endpoint. Instead, it is sufficient to check that the identifier of the other endpoint in context is not equal to any of one’s own identifiers. To simplify implementation of this check, it is RECOMMENDED that each endpoint selects one globally unique identifier and uses it in all PSK handshakes. The unique identifier can, for example, be one of its MAC addresses or a 32-byte random number. Including node identifiers in the ImportedIdentity.context construct also ensures that condition (2) is met.
```

Comment:
This is redundant with the text in I-D.ietf-tls-external-psk-importer. Since item (3) already recommends using it, I think we should remove this entire bullet.

Reply:
This is a guidance document so it is worth pointing out different ways of doing things. I added a reference to UUID (RFC 4122) based on feedback from Jonathan. I understand that you want to keep the recommendations concise. So moved this text to the security considerations.